### ISSUE TITLE
Adding an opportunity too add model specific losses

### ISSUE THREAD

**Issue by StillerPatrick** [ComputationalRadiationPhysics/NeuralSolvers#17]
Models like the GatedPINN uses a model specific loss in order to balance the utilization of the experts. 
An option would be to give the model an attribute called loss. 

In the PINN loss we can call the loss function with :
```python
if hasattr(self.model, 'loss'):
    pinn_loss = pinn_loss + model.loss
```
